cpp/tests/reductions/reduction_tests.cu

```diff
@@ -756,4 +755,125 @@ TEST_F(StringReductionTest, AllNull)
   EXPECT_FALSE(maxresult->is_valid());
 }
 
+TYPED_TEST(ReductionTest, Count)
+{
+    using T = TypeParam;
+    std::vector<int> int_values({6, -14, 13, 64, 0, -13, -20, 45});
+    std::vector<bool> host_bools({1, 1, 0, 0, 1, 1, 1, 1});
+    std::vector<T> v = convert_values<T>(int_values);
+
+    // test without nulls
+    cudf::test::fixed_width_column_wrapper<T> col(v.begin(), v.end());
+    cudf::size_type expected_value = v.size();
+    this->reduction_test(col, expected_value, this->ret_non_arithmetic, cudf::experimental::make_count_aggregation(cudf::include_nulls::YES));
+    this->reduction_test(col, expected_value, this->ret_non_arithmetic, cudf::experimental::make_count_aggregation(cudf::include_nulls::NO));
+
+    // test with nulls
+    cudf::test::fixed_width_column_wrapper<T> col_nulls = construct_null_column(v, host_bools);
+    cudf::size_type valid_count = cudf::column_view(col_nulls).size() - cudf::column_view(col_nulls).null_count();
+    auto r = replace_nulls(v, host_bools, T{0});
+    cudf::size_type expected_null_value = std::accumulate(host_bools.begin(), host_bools.end(), 0);
+
+    this->reduction_test(col_nulls, expected_value, this->ret_non_arithmetic,
+        cudf::experimental::make_count_aggregation(cudf::include_nulls::YES));
+    this->reduction_test(col_nulls, expected_null_value, this->ret_non_arithmetic,
+        cudf::experimental::make_count_aggregation(cudf::include_nulls::NO));    
+}
+
+TYPED_TEST(ReductionTest, Median)
```

Comment:
Might be good to add a test with odd number of elements, since the median calculation is different in that case.

Reply:
added.